### PR TITLE
[addons] Add getWatchedMode plugin API and constants

### PR DESCRIPTION
**Pull request by dbrobins**
## Description
Provides an API that plugins can used to get the show-watched mode (unwatched, watched, all) and related constants.

## Motivation and context
My plugin provides TV shows based on an external database. It changes behavior based on watched mode.

## How has this been tested?
Verified by invoking the API from the plugin and logging/using the reported watched mode and ensuring it matches with what was set in the GUI. Adding an API should not directly affect other areas of the code. Tested in a Linux environment.

## What is the effect on users?
Provides a new API. Added inline documentation describing the API.

## Types of change
- [ ] **Bug fix** (non-breaking change which fixes an issue)
- [ ] **Clean up** (non-breaking change which removes non-working, unmaintained functionality)
- [ ] **Improvement** (non-breaking change which improves existing functionality)
- [X] **New feature** (non-breaking change which adds functionality)
- [ ] **Breaking change** (fix or feature that will cause existing functionality to change)
- [ ] **Cosmetic change** (non-breaking change that doesn't touch code)
- [ ] **None of the above** (please explain below)

## Checklist:
- [X] My code follows the **[Code Guidelines](https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md)** of this project 
- [X] My change requires a change to the documentation, either Doxygen or wiki
- [X] I have updated the documentation accordingly
- [X] I have read the **[Contributing](https://github.com/xbmc/xbmc/blob/master/docs/CONTRIBUTING.md)** document
- [ ] I have added tests to cover my change
- [ ] All new and existing tests passed

### CODE REVIEW

File: xbmc/interfaces/legacy/ModuleXbmcplugin.h [hunk 1]
```diff
@@ -438,6 +439,33 @@ namespace XBMCAddon
     void setProperty(int handle, const char* key, const String& value);
 #endif
 
+#ifdef DOXYGEN_SHOULD_USE_THIS
+    ///
+    /// \ingroup python_xbmcplugin
+    /// @brief \python_func{ xbmcplugin.getWatchedMode(handle, content) }
+    ///-------------------------------------------------------------------------
+    /// Get the watched mode for given content type:
```
Comment (CastagnaIT) [inline xbmc/interfaces/legacy/ModuleXbmcplugin.h:447]: the description is not well explained,  get watched mode for what?
this get the current filter mode of current directory

and the type of return value must be placed in `@return` key description

File: xbmc/interfaces/legacy/ModuleXbmcplugin.h [hunk 2]
```diff
@@ -438,6 +439,33 @@ namespace XBMCAddon
...
+    /// ~~~~~~~~~~~~~{.py}
+    /// ..
+    /// xbmcplugin.getWatchedMode(int(sys.argv[1]), 'tvshows')
+    /// ..
+    /// ~~~~~~~~~~~~~
+    ///
+    getWatchedMode(...);
+    ///@}
+#else
+    int getWatchedMode(int handle, const char* content = NULL);
```
Comment (CastagnaIT) [inline xbmc/interfaces/legacy/ModuleXbmcplugin.h:466]: if you mean directory IMO `xbmcplugin.getWatchedMode` is not the best method name, should be something like `xbmcplugin.getDirectoryWatchedMode`

**Comment by lrusak** [comment]
Uhhhh? What just happened? Failed rebase? Wrong branch?

**Comment by dbrobins** [comment]
Failed rebase.

**Pull request closed without merging**